process/block/metablock.go

```diff
@@ -2017,6 +2129,13 @@ func (mp *metaProcessor) createShardInfo() ([]data.ShardDataHandler, error) {
 			continue
 		}
 
+		isBlockAfterEquivalentMessagesFlag := check.IfNil(headerInfo.hdr) &&
```

Comment:
for the shardInfo it seems we are missing some validation
This should be done on the `func (imh *InterceptedMetaHeader) CheckValidity()` -> here we should check at least that all required relevant fields are filled.

And also in checkShardHeadersValidity (during processing on metachain) -> here there should be checked the existance of proofs (and the correctness)

Reply:
added basic verification for CheckValidity and also on checkShardHeadersValidity